packages/manager/apps/pci-block-storage/src/pages/edit/Edit.page.tsx

```diff
@@ -82,16 +77,24 @@ export default function EditPage() {
 
   const projectUrl = useProjectUrl('public-cloud');
   const { data: project } = useProject(projectId || '');
-  const catalogPrice = useGetPrices(projectId, volumeId);
   const { translateMicroRegion } = useTranslatedMicroRegions();
 
-  const { getTextPrice } = useCatalogPrice(3);
-
   const {
     data: volume,
     isLoading: isLoadingVolume,
     isPending: isPendingVolume,
   } = useVolume(projectId, volumeId);
+  const { data: catalog } = useCatalog();
+
+  const catalogVolume = useMemo(() => {
+    if (!!catalog && !!volume) {
+      return (
+        catalog.addons.find((addon) => addon.planCode === volume.planCode) ||
+        null
+      );
+    }
+    return null;
+  }, [catalog, volume]);
```

Comment:
Maybe we can simplify a bit, by returning explicit `undefined` value:
```suggestion
  const catalogVolume = useMemo(
    (): TAddon | undefined =>
      catalog?.addons.find((addon) => addon.planCode === volume?.planCode),
    [catalog, volume],
  );
```

Reply:
Doing this will search through catalog even if the volume isn't fetch